### ISSUE TITLE
How to implement theme selector inside a react app?

### ISSUE THREAD

**Issue by zagoorland** [FIameCaster/prism-react-editor#1]
Hey. I'm struggling to implement theme switcher using this library.
Here's my code snippet:

```tsx
import { Editor } from "prism-react-editor"
import { BasicSetup } from "prism-react-editor/setups"
import "prism-react-editor/prism/languages/jsx"
import "prism-react-editor/prism/languages/tsx"
import "prism-react-editor/languages"

import "prism-react-editor/layout.css"
// import "prism-react-editor/themes/github-dark.css"
// import "prism-react-editor/themes/github-light.css"
import { loadTheme } from "prism-react-editor/themes"

// Required by the basic setup
import "prism-react-editor/search.css"
import { useEffect, useLayoutEffect } from "react"

interface CodeEditorProps {
  language: string
  theme: string
}

export default function CodeEditor({ language, theme }: CodeEditorProps) {
  useEffect(() => {
    loadTheme("github-dark")
  }, [])

  return (
    <main className="grow overflow-auto github">
      <Editor
        language={language}
        value={`const foo: string = 'bar';\nfunction greet(name: string): string {\n  return 'Hello, ' + name;\n}\nconsole.log(greet('World'));\n${"console.log(foo);\n".repeat(100)}`} // Long content to test scrolling
        style={{
          height: "100%",
          fontSize: "16px",
        }}
      >
        {(editor) => <BasicSetup editor={editor} />}
      </Editor>
    </main>
  )
}

```

Using this example styles are not working at all :( 

Any tips how to get it working? It would be nice to have some example inside readme :)

Cheers!

**Comment by jonpyt** [comment]
`loadTheme()` does not inject the theme's CSS into the page, it simply returns a Promise that resolves with the theme's CSS. You need to add the CSS returned to a `<style>` element. This could be achieved with an example like this:

```tsx
export default function CodeEditor({ language, theme }: CodeEditorProps) {
  const [themeCss, setCss] = useState<string | null>(null)

  useEffect(() => {
    let cancelled = false
    loadTheme(theme).then(css => {
      if (!cancelled) setCss(css)
    })
    return () => cancelled = true
  }, [theme])

  return (
    themeCss && <main className="grow overflow-auto github">
      <style>{themeCss}</style>
      <Editor
        language={language}
        value="const foo = 'bar'"
        style={{
          height: "100%",
          fontSize: "16px",
        }}
      >
        {(editor) => <BasicSetup editor={editor} />}
      </Editor>
    </main>
  )
}
```

Note that you'll get multiple `<style>` elements with themes on the page if the `CodeEditor` component is used more than once. If this is the case, you should move the theme and the `<style>` element to a higher component that's only used once per page.

The editor also won't be visible before the theme loads. There are ways around this, but they're cumbersome. You can render the editor even if `themeCss` is null, but this will result in an [FOUC](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) before the theme loads. It's up to you to decide what's best.

Hope this helps!

**Comment by zagoorland** [comment]
It helped! It works perfect now :)
There's no way I would solve it by myself, there's no much in the docs about usage of loadTheme function. Please, consider adding some example with select element and useTheme :)

Thank you :)

**Comment by jonpyt** [comment]
There is an example showing the `loadTheme` function at the [bottom of the readme](https://github.com/FIameCaster/prism-react-editor?tab=readme-ov-file#themes) and the function is documented with JSDoc, which you should be able to read in your IDE.

> Please, consider adding some example with select element and useTheme :)

Adding a `useTheme` hook to `prism-react-editor/themes` is a good idea. I will consider this.

**Comment by Seanw265** [comment]
I agree that the documentation for loadTheme could be clearer. For example, the JSDoc description (`Asynchronously loads the theme with the specified name.`) doesn’t explain whether it just fetches the CSS or applies the styles directly.

This is a great project, and adding a quick example or more detail in the docs could make it even more user-friendly. Thanks for your hard work—I’d love to see this project grow!

**Comment by jonpyt** [comment]
I will specify that `loadTheme` returns CSS that must be added to a `<style>` tag on the page in the JSDoc comment.

I'm also planning a making a `useTheme` hook that can be used to make a theme switcher more easily.
